tests/microos/workloads/kea-container/dhcp4_client.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE"s openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Package: podman kea-container
+# Summary: install and verify Kea DHCP server container.
+# Maintainer: QE Core <qe-core@suse.de>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use lockapi;
+use Utils::Systemd qw(disable_and_stop_service systemctl check_unit_file);
+
+sub run {
+    my ($self) = @_;
+    mutex_wait 'barrier_setup_done';
+
+    assert_script_run("ip a");
+
+    mutex_wait 'barrier_setup_done';
+    barrier_wait 'DHCP_SERVER_READY';
+
+    # Configure static network, disable firewall
+    disable_and_stop_service($self->firewall) if check_unit_file($self->firewall);
+
+    assert_script_run('nmcli conn');
+    my $nm_list = script_output("nmcli -t -f DEVICE,NAME c | head -n1");
+    my ($device, $nm_id) = split(':', $nm_list);
+    assert_script_run "nmcli connection modify '$nm_id' ipv4.method auto";
+    assert_script_run("nmcli conn up '$nm_id'");
+    assert_script_run('nmcli conn');
+
+    assert_script_run("ip a");
+    assert_script_run('ping -c 1 10.0.2.101');
+
+    $nm_list = script_output("nmcli -t -f DEVICE,NAME c | head -n1");
+    ($device, $nm_id) = split(':', $nm_list);
+    validate_script_output("ip -4 addr show dev $device | sed -rne '/inet/s/[[:blank:]]*inet ([0-9\\.]*).*/\\1/p'", sub { m/10.0.2.12[0-5]/ });
```

Comment:
The regex in `sub` should IMHO match the whole used subnet above `"subnet": "10.0.2.0/24",`, no?

Reply:
I am verifying that the IP address falls within the range of defined pools, rather than the subnet. Would you like me to match the subnet instead?

Reply:
Ah, I didn't saw the `pools` section before, only the subnet definition. Then this is correct and nothing to be done